execution/executor/src/lib.rs

```diff
@@ -931,12 +920,6 @@ impl<V: VMExecutor> BlockExecutor for Executor<V> {
             )?;
         }
 
-        // Prune the tree.
-        for block in blocks {
-            for txn_data in block.output().transaction_data() {
-                txn_data.prune_state_tree();
-            }
-        }
```

Comment:
@wqfish  @lightmark  remind me why we need to prune it manually in the first place? What's wrong with waiting for them to drop? And does the same reason make a `prune()` call necessary even we now (with this change I mean) keep only one speculative smt for the block?

Reply:
Thanks for asking this question for me. This is the piece of code where I'm not so certain about! I suppose it's because we didn't prune the last node yet? So instead we should call `prune_state_tree` in the `ExecutedTrees`?

Reply:
Because we don't want to have a huge number of nodes in scratchpad except for all that can be reached by the last root hash. if batch update is supported this call will be never useful I think.

Reply:
We may have committed only some of the blocks that have executed, right? If so, then I guess by pruning here we were saving space on the nodes that carry states that were committed (if needed, we will fetch them by inclusion/non-inclusion proofs and they will come from storage with the right state), while still keeping the speculative states from blocks that were executed but not committed.